### PR TITLE
Add templating for MQTT topics

### PR DESCRIPTION
**Pull request by fhriley**
This adds the ability to use templates when specifying the MQTT
topic. This is much more useful than static topics since you
can create topics like this:

switches/{{ index .Tags "agent_host" }}/ports/{{ index .Tags "port_num" }}

###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by fhriley** [comment]
Is anyone looking at these?

**Comment by fhriley** [comment]
@nathanielc @aanthony1243 Can someone look at this? We are using this in production, but we prefer not to have to maintain a custom build of kapacitor. The static topics that kapacitor supports without this change are essentially useless.

File: services/mqtt/service.go [hunk 1]
```diff
@@ -233,11 +242,81 @@ type handler struct {
...
+
+	// Alert Level, one of: INFO, WARNING, CRITICAL.
+	Level string
+
+	// Time
+	Time time.Time
+
+	// Duration of the alert
+	Duration time.Duration
+}
```
Comment (nathanielc) [inline services/mqtt/service.go]: These structures should not be copied. Use the `TemplateData` function on the alert Event in order to use consistent template data for the templates.

See https://github.com/influxdata/kapacitor/blob/master/services/pagerduty2/service.go#L402 as an example.

File: services/mqtt/service.go [hunk 2]
```diff
@@ -233,11 +242,81 @@ type handler struct {
...
+		Duration: event.State.Duration,
+	}
+
+	topicBuf := h.s.bufPool.Get().(*bytes.Buffer)
+	defer func() {
+		topicBuf.Reset()
+		h.s.bufPool.Put(topicBuf)
+	}()
+
+	topicTmpl, err := text.New("topic").Parse(topic)
```
Comment (nathanielc) [inline services/mqtt/service.go]: The topic template doesn't change for a given handler, can you udpate this to parse the template during handler construction? Then you can eval the template for each eval without parsing it. This also provides the user with earlier feedback if they have a typo in their template.

File: services/mqtt/service.go (hunk 1 above)
Comment (fhriley) [reply services/mqtt/service.go]: Done.

File: services/mqtt/service.go (hunk 2 above)
Comment (fhriley) [reply services/mqtt/service.go]: Done.

**Comment by nathanielc** [comment]
@fhriley LGTM, can you fix the conflicts and then I can merge.

**Comment by fhriley** [comment]
Merge conflict resolve.d

**Comment by fhriley** [comment]
Test added.

**Pull request merged**